src/components/molecules/ScheduleEvent/ScheduleEvent.tsx

```diff
@@ -81,26 +122,103 @@ export const ScheduleEvent: React.FC<ScheduleEventProps> = ({
 
   const onEventBoxClick = useCallback(
     (e) => {
-      if (e.target.closest(`.${ScheduleEventBookmarkClass}`)) return;
+      if (
+        e.target.closest(`.${ScheduleEventBookmarkClass}`) ||
+        e.target.closest(`.${ScheduleEventExpandClass}`)
+      )
+        return;
 
       showEventModal();
     },
     [showEventModal]
   );
 
+  const handleMouseOver = useCallback(() => {
+    if (isMinimallyLongEvent) return;
+
+    showExpanded();
+  }, [isMinimallyLongEvent, showExpanded]);
+
+  const handleMouseOut = useCallback(() => {
+    if (isExpandedStatic) return;
+
+    hideExpanded();
+  }, [isExpandedStatic, hideExpanded]);
+
+  const handleCollapse = useCallback(() => {
+    hideExpandedStatic();
+    hideExpanded();
+  }, [hideExpanded, hideExpandedStatic]);
+
+  const handleToggleExpand = useCallback(() => {
+    if (!isExpandedStatic) {
+      toggleExpandedStatic();
+
+      return;
+    }
+
+    handleCollapse();
+  }, [isExpandedStatic, handleCollapse, toggleExpandedStatic]);
+
+  useEffect(() => {
+    const handleClickOutside = (event: Event) => {
+      const eventTarget = event.target as Node;
+      if (
+        eventRef.current &&
+        !eventRef?.current?.contains(eventTarget) &&
+        !isEventModalVisible
+      ) {
+        handleCollapse();
+      }
+    };
+
+    window.addEventListener("mousedown", handleClickOutside, false);
+
+    return () => {
+      window.removeEventListener("mousedown", handleClickOutside, false);
+    };
+  }, [eventRef, handleCollapse, isEventModalVisible]);
```

Comment:
Why do we need to use `window.addEventListener` here? Modals already handle closing when clicked outside of..?

Reply:
I'm not convinced this entire `useEffect` is valid/needed here.

Reply:
probably I need to add that `eventListener` to another node, but overall logic should stay
https://github.com/sparkletown/sparkle/pull/1733#discussion_r665959998

Reply:
removed 2c9b3bd